code-challenges.test.js

```diff
@@ -10,42 +10,225 @@
 // Reminder: The test will call your function
 // Run the file with the following command: $ yarn jest
 
-// --------------------1) Create a function that takes in an array of objects and returns an array with a sentence about each person with their name capitalized.
+
+
+
+// --------------------1) 
+// Create a function that takes in an array of objects and returns an array with a sentence about each person with their name capitalized.
+
 
 // a) Create a test with an expect statement using the variable provided.
 
-const people = [
-  { name: "ford prefect", occupation: "a hitchhiker" },
-  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+
+describe("aboutPeople", (array) => {
+  it("takes in an array of objects and returns an array with a sentence about each person with their name capitalized", () => {
+    const people = [
+      { name: "ford prefect", occupation: "a hitchhiker" },
+      { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+      { name: "arthur dent", occupation: "a radio employee" }
+    ]
+      expect(aboutPeople(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
+
+
+// ******* RED FAILURE *******
+// ● aboutPeople › takes in an array of objects and returns an array with a sentence about each person with their name capitalized
+// ReferenceError: aboutPeople is not defined
+
+
 
 
 // b) Create the function that makes the test pass.
 
 
-// --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
+
+// ******* PSUEDOCODE *******
+// input = array of objects
+  // map over array of objects by referencing the keys
+  // .split(" ") --> splits name key into 2 elements
+    // .toUpperCase() the zero index of each element
+  // Because of the split only the zero index, or first letter will print
+    // .substring(1) to print the remainder of the word beginning with index 1
+// output = array of sentences containing the capitalized names and lowercase occupation
+  
+
+
+// Attempt #940523 (the closest "dynamic" solution without being the solution)
```

Comment:
This problem is so tricky to get to be dynamic!  Don't beat yourself up over it.  Best way to get there is a nested map which can get mind bendy if you let it.  Great job for pushing yourself though!